CONTRIBUTING.md

```diff
@@ -22,6 +23,30 @@ This project has adopted the [Microsoft Open Source Code of Conduct](https://ope
 kubectl scale deployments/inflate --replicas=3
 ```
 
+### Use local environment instead of GitHub Codespaces. 
+> Tested environment : WSL2
+
+1. **Install tools**
+   * make
+   * kubectl
+   * [helm](https://github.com/helm/helm/releases)
+   * [golang](https://go.dev/dl/) > 1.21
+   * [yq](https://github.com/mikefarah/yq/releases) / [jq](https://github.com/jqlang/jq/releases)
+   * [skaffold](https://skaffold.dev/docs/install/#standalone-binary)
+   * [azure-cli](https://learn.microsoft.com/en-us/cli/azure/install-azure-cli)
+
+2. **Provision cluster, build and deploy Karpenter** : Set `AZURE_SUBSCRIPTION_ID` to your subscription (and customize region in `Makefile-az.mk` if desired). Then run `make az-all`. This logs into Azure (follow the prompts), provisions AKS and ACR (using resource group `$COMMON_NAME`, COMMON_NAME have to be unique), builds and deploys Karpenter, deploys sample `default` Provisioner and `inflate` Deployment workload.
+```
+az config set core.output=json
+export AZURE_SUBSCRIPTION_ID=xxxx-xxxx-xxxx-xxxx
+export COMMON_NAME=karpenter2test
```

Comment:
some notes on teh semantics that are accepted by common name are important. For ACR it will not allow dashes etc. Maybe mentioning this might be useful?

Reply:
Thanks for your kindly advice.
I made a change. i think your example is good, so i added it